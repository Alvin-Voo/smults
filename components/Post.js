import React from 'react';
import PropTypes from 'prop-types';
import {
  Item, Icon, Label,
} from 'semantic-ui-react';

export default function Post(props) {
  const { post } = props;
  const {
    image, title, author, category, created, body,
    pending_payout_value, net_votes, children, tags, url,
  } = post;

  return (
    <Item href={url} target="_blank">
      <Item.Image size="small" src={image} />
      <Item.Content>
        <Item.Header>{title}</Item.Header>
        <Item.Meta>
          <Label color="purple">{author}</Label>
          {' '}
          in
          {' '}
          <Label color="teal" tag>{category}</Label>
          {' '}
          â€¢
          {' '}
          {created}
        </Item.Meta>
        <Item.Description>
          {body}
        </Item.Description>
        <Item.Extra>
          <Icon name="dollar" />
          {pending_payout_value}
          <Icon name="sort up" />
          {net_votes}
          <Icon name="comment outline" />
          {children}
          <span style={{ margin: '0 .5em' }} />
          {
              tags.map(tag => <Label key={tag}>{tag}</Label>)
            }
        </Item.Extra>
      </Item.Content>
    </Item>
  );
}

Post.propTypes = {
  post: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number,
    PropTypes.arrayOf(PropTypes.string)])).isRequired,
};
