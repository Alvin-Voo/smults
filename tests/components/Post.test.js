import React from 'react';
import { shallow } from 'enzyme';
import Post from '../../components/Post';
import postListData from '../fixtures/postList';

test('should render a single post by author correctly', () => {
  const wrapper = shallow(<Post post={postListData[0]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render a \'resteem\' label when filtering by author and encounter resteemded post', () => {
  const wrapper = shallow(<Post post={postListData[1]} />);
  expect(wrapper.find('Label').at(0).find('Icon').prop('name')).toBe('share');
  expect(wrapper.find('Label').at(0).html()).toEqual(expect.stringContaining('resteemed'));
  expect(wrapper).toMatchSnapshot();
});
