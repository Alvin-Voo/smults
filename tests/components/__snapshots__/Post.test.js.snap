// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a 'resteem' label when filtering by author and encounter resteemded post 1`] = `
<Item
  href="http://steemit.com/some-permlink-too"
  target="_blank"
>
  <ItemImage
    size="small"
    src="http://image2.com"
  />
  <ItemContent>
    <ItemHeader>
      this is a title too!
    </ItemHeader>
    <ItemMeta>
      <Label>
        <Icon
          as="i"
          name="share"
        />
        resteemed
      </Label>
      <Label
        color="purple"
      >
        dummy2
      </Label>
       
      in
       
      <Label
        color="teal"
        tag={true}
      >
        sun
      </Label>
       
      •
       
      6 hours ago
    </ItemMeta>
    <ItemDescription>
      text body2
    </ItemDescription>
    <ItemExtra>
      <Icon
        as="i"
        name="dollar"
      />
      105.001 SBD
      <Icon
        as="i"
        name="sort up"
      />
      2
      <Icon
        as="i"
        name="comment outline"
      />
      12
      <span
        style={
          Object {
            "margin": "0 .5em",
          }
        }
      />
      <Label
        key="0"
      >
        tag4
      </Label>
      <Label
        key="1"
      >
        tag5
      </Label>
      <Label
        key="2"
      >
        tag6
      </Label>
    </ItemExtra>
  </ItemContent>
</Item>
`;

exports[`should render a single post by author correctly 1`] = `
<Item
  href="http://steemit.com/some-permlink"
  target="_blank"
>
  <ItemImage
    size="small"
    src="http://image.com"
  />
  <ItemContent>
    <ItemHeader>
      this is a dummy title
    </ItemHeader>
    <ItemMeta>
      <div />
      <Label
        color="purple"
      >
        dummy
      </Label>
       
      in
       
      <Label
        color="teal"
        tag={true}
      >
        moon
      </Label>
       
      •
       
      5 hours ago
    </ItemMeta>
    <ItemDescription>
      text body
    </ItemDescription>
    <ItemExtra>
      <Icon
        as="i"
        name="dollar"
      />
      65.012 SBD
      <Icon
        as="i"
        name="sort up"
      />
      213
      <Icon
        as="i"
        name="comment outline"
      />
      24
      <span
        style={
          Object {
            "margin": "0 .5em",
          }
        }
      />
      <Label
        key="0"
      >
        tag1
      </Label>
      <Label
        key="1"
      >
        tag2
      </Label>
      <Label
        key="2"
      >
        tag3
      </Label>
    </ItemExtra>
  </ItemContent>
</Item>
`;
